### PR TITLE
for [#33] Ride hailing company design and initial implementation

### PR DESCRIPTION
**Pull request by dserdyuk**

### CODE REVIEW

**Comment by colinsheppard** [comment]
Looks good so far. I will have time for a more detailed review tomorrow.

File: src/main/scala/beam/agentsim/agents/modalBehaviors/ChoosesMode.scala [hunk 1]
```diff
@@ -72,7 +72,7 @@ trait ChoosesMode extends BeamAgent[PersonData] with TriggerShortcuts with HasSe
     case Event(TriggerWithId(FinalizeModeChoiceTrigger(tick), theTriggerId), info: BeamAgentInfo[PersonData]) =>
       hasReceivedCompleteChoiceTrigger = true
       val result : State = completeChoiceIfReady(Some(theTriggerId))
-      if(result.stateName == ChoosesMode){
+      if(result.stateName == ChoosingMode){
```
Comment (dserdyuk) [inline src/main/scala/beam/agentsim/agents/modalBehaviors/ChoosesMode.scala]: @colinsheppard There is an attempt to compare unrelated types. Probably, misprint

**Comment by dserdyuk** [comment]
@colinsheppard Alright.. Inquiry -> Reservation -> Pickup use case is ready as for me. Please review changes.. I guess next logical step is to implement DrivesVehicle

**Comment by dserdyuk** [comment]
@colinsheppard  Enter/LeaveVehicle event handling and BeamVehicleAgent convertion to BeamAgent have been added

File: src/main/resources/beam-template.conf [hunk 2]
```diff
@@ -85,6 +95,12 @@ beam.events.pathTraversalEvents = ["car","bus","subway","tram","rail","car","wal
 beam.events.filterDist = 10000.0
 
 ##################################################################
+# Ride Hailing Params
+##################################################################
+beam.taxi.defaultCostPerMile=1.25
```
Comment (dserdyuk) [inline src/main/resources/beam-template.conf]: just random value

File: src/main/scala/beam/agentsim/agents/TaxiAgent.scala [hunk 3]
```diff
@@ -66,7 +66,7 @@ class TaxiAgent(override val id: Id[TaxiAgent], override val data: TaxiData, val
 
   when(Traveling) {
```
Comment (dserdyuk) [inline src/main/scala/beam/agentsim/agents/TaxiAgent.scala]: @colinsheppard I'm thinking to fuse TaxiAgent and BeamVehicleAgent

File: src/main/scala/beam/agentsim/agents/modalBehaviors/ChoosesMode.scala [hunk 4]
```diff
@@ -53,17 +53,17 @@ trait ChoosesMode extends BeamAgent[PersonData] with TriggerShortcuts with HasSe
     case Event(TriggerWithId(BeginModeChoiceTrigger(tick), triggerId), info: BeamAgentInfo[PersonData]) =>
       logInfo(s"inside ChoosesMode @ ${stateData.tick}")
       val nextAct = stateData.data.nextActivity.right.get // No danger of failure here
-      services.beamRouter ! RoutingRequest(stateData.data.currentActivity, nextAct, DiscreteTime(stateData.tick.get.toInt), Vector(BeamMode.CAR, BeamMode.BIKE, BeamMode.WALK, BeamMode.TRANSIT), id)
+      beamServices.beamRouter ! RoutingRequest(stateData.data.currentActivity, nextAct, DiscreteTime(stateData.tick.get.toInt), Vector(BeamMode.CAR, BeamMode.BIKE, BeamMode.WALK, BeamMode.TRANSIT), id)
       //TODO parameterize search distance
-      services.taxiManager ! TaxiInquiry(stateData.data.currentActivity.getCoord, 2000)
+      //services.taxiManager ! TaxiInquiry(info.id, stateData.data.currentActivity,  2000)
```
Comment (dserdyuk) [inline src/main/scala/beam/agentsim/agents/modalBehaviors/ChoosesMode.scala]: need to remember why I comment this

**Comment by dserdyuk** [comment]
Merged with agentsim branch

**Comment by dserdyuk** [comment]
@colinsheppard Some fixes and improvment. I guess we could merge this changes because it takes me more and more time to merge between agentsim and my branch

**Pull request merged**